File: backend/app/schemas/student.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel


class StudentCreate(BaseModel):
    rut: str | None = None
    first_name: str
    last_name: str
    date_of_birth: date | None = None
    grade: str | None = None
    school_name: str | None = None
    diagnosis: str | None = None
    needs: list[str] = []
    assigned_educator_id: uuid.UUID | None = None


class StudentUpdate(BaseModel):
    rut: str | None = None
    first_name: str | None = None
    last_name: str | None = None
    date_of_birth: date | None = None
    grade: str | None = None
    school_name: str | None = None
    diagnosis: str | None = None
    needs: list[str] | None = None
    assigned_educator_id: uuid.UUID | None = None


class StudentOut(BaseModel):
    id: uuid.UUID
    rut: str | None
    first_name: str
    last_name: str
    date_of_birth: date | None
    grade: str | None
    school_name: str | None
    diagnosis: str | None
    needs: list[str]
    assigned_educator_id: uuid.UUID | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/student.ts
export interface Student {
  id: string;
  rut: string | null;
  first_name: string;
  last_name: string;
  date_of_birth: string | null;
  grade: string | null;
  school_name: string | null;
  diagnosis: string | null;
  needs: string[];
  assigned_educator_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface StudentCreate {
  rut?: string;
  first_name: string;
  last_name: string;
  date_of_birth?: string;
  grade?: string;
  school_name?: string;
  diagnosis?: string;
  needs?: string[];
}

export interface StudentUpdate extends Partial<StudentCreate> {}
